paymaster-analysis/alchemy/alchemy-account.ts

```diff
@@ -0,0 +1,290 @@
+import dotenv from "dotenv";
+import { getAccountNonce } from "permissionless";
+import { UserOperation, signUserOperation } from "./utils/userOp";
+import { Hash, createPublicClient, http } from "viem";
+import { privateKeyToAccount } from "viem/accounts";
+import { goerli, sepolia } from "viem/chains";
+import {
+  SAFE_ADDRESSES_MAP,
+  encodeCallData,
+  getAccountAddress,
+  getAccountInitCode,
+} from "./utils/safe";
+import { setTimeout } from "timers/promises";
+
+dotenv.config();
+const privateKey = process.env.PRIVATE_KEY;
+const ENTRY_POINT_ADDRESS = process.env.ALCHEMY_ENTRYPOINT_ADDRESS;
+const multiSendAddress = process.env.ALCHEMY_MULTISEND_ADDRESS;
+const saltNonce = BigInt(process.env.ALCHEMY_ACCOUNT_PAYMASTER_NONCE);
+const chain = process.env.ALCHEMY_CHAIN;
+const chainID = Number(process.env.ALCHEMY_CHAIN_ID);
+const safeVersion = process.env.SAFE_VERSION;
+const rpcURL = process.env.ALCHEMY_RPC_URL;
+const policyID = process.env.ALCHEMY_GAS_POLICY;
+const apiKey = process.env.ALCHEMY_API_KEY;
```

Comment:
I see many TS errors due to the lack of env variable validation (they can be undefined): 
![Screenshot 2023-12-11 at 17 16 30](https://github.com/safe-global/safe-modules/assets/16622558/0dab520f-f008-44ad-9d94-b10a6b903b04)

Reply:
Done 👍🏾

Reply:
Looks like there's one still in the `utils/userOps.ts` file:
```
~/projects/safe/safe-modules/4337-gas-metering gas-metering ❯ npx tsc --noEmit                                                                                                                               14:31:32
utils/userOps.ts:76:32 - error TS2538: Type 'undefined' cannot be used as an index type.

76             SAFE_ADDRESSES_MAP[safeVersion][chainID].SAFE_4337_MODULE_ADDRESS,
                                  ~~~~~~~~~~~


Found 1 error in utils/userOps.ts:76
```

Reply:
This one was solved in commit: 9034d2b1c3f6712f37efb40e1b58c50d3b496d44, forgot to push. Sorry!